Content.Server/Radio/EntitySystems/RadioSystem.cs

```diff
@@ -255,6 +256,9 @@ public void SendRadioMessage(
             // Check if within range for range-limited channels
             if (channel.MaxRange.HasValue && channel.MaxRange.Value > 0)
             {
+                if (transform.MapID != sourceMapId)
```

Comment:
I am not sure if other conditions should apply.
```suggestion
                if (!channel.LongRange && transform.MapID != sourceMapId  && !radio.GlobalReceive)
```